### ISSUE TITLE
Enhancement request: Media Browser functionality

### ISSUE THREAD

**Issue by B-Hartley** [RogerSelwyn/Home_Assistant_SkyQ_MediaPlayer#24]
Have you considered using the new media browser functionality to allow channel selection through the media player with this SkyQ integration?

If you wanted to go crazy, could it even browse recordings on the box.

Just dreaming I guess !

Thanks

**Comment by RogerSelwyn** [comment]
I did think about it then discard it because you can't cause a recording to play. But it is possible to play channels, so I'll think about it again.  I could enable browsing of recordings, but it isn't possible to make them play, so not really sure of the point of it, which is why I discarded that idea.

**Comment by RogerSelwyn** [comment]
@B-Hartley - I'm interested in your views here. 

As I think about it I could provide information to the media browser that showed the list of channels. To provide the current programme on each channel though is likely to have very poor performance - I basically have to go and get the programme Information for each channel (something I already know is poor) and to do that across 100's of channels is impractical. 

I could do it just for the configured list of channels that appear on the media player source drop down. It would provide another way of selecting a channel, but not really any major new functionality. It's something I'll think about some more.

Listing recordings whilst fairly easy (I would think) wouldn't make them playable I'm afraid.

**Comment by B-Hartley** [comment]
That all makes sense.
I would say if you could provide a "what's on now" for the channels that are "favourited" by being listed on the device, then that would be great.
I agree, it's much more a cool nice to have feature, than anything that truly adds value, but it would definitely look cool.

Thanks

**Comment by RogerSelwyn** [comment]
@B-Hartley I've posted an alpha release (which should be pretty stable), which will show channel images for configured channels. 

It will have a go at custom sources, but the name needs to match the name in the EPG (and if it does why use a custom source instead of a channel source). I've just thought of a way to maybe improve on some of that detection as well, so I'll try to do that over the next couple of days. 

For things like Netflix (which can only be done via a custom source), it will try to find out if there is an app image in the www folder that matches the name of source.

**Comment by B-Hartley** [comment]
Looks like a good start. 
Do you have any control over the aspect ratio of the images ?

**Comment by RogerSelwyn** [comment]
I can configure to use a different url if I can find one, but the logos at that url are a set aspect ratio (which I can change the size of, but that won't change how they fit :-(.

**Comment by RogerSelwyn** [comment]
@B-Hartley - I've now changed to use live programme image and show the current programme in the title. If you have multiple  channels set up you should see how this slows down. Let me know what you think. I can always at a config option to allow people to disable live images.

**Comment by B-Hartley** [comment]
That looks good.
I have about 11 channels on my lineup.
So the delay is noticeable, but acceptable I think.
Is there any cacheing of the informaion / images ?
or does it call all the images up everytime you press the browser ?

**Comment by RogerSelwyn** [comment]
Caching of images isn't really the issue. The issue is pulling down the EPG for each channel for the current day and then identifying the current programme. This is quite a lot of data since for the UK, you can only pull a full day.

This is an example of a set of data for BBC1 (I think). So think of this for x channels. Not sure if I can do the queries in parallel, I'll see if that can be done.

**Comment by RogerSelwyn** [comment]
It now caches unto 20 channels, so it should be slow for the first time each day (or possibly two days because I think I fetch two days EPG for reasons I won't go into here).

**Comment by B-Hartley** [comment]
Fab !  I'll give it a go later.

**Comment by B-Hartley** [comment]
That's brilliant, pretty much instant !

**Comment by B-Hartley** [comment]
Are you matching the channels by their channel number?
If I use a custom source list to put the channels in a sensible order and use the favourites to select them, it doesn't work.
Not big deal, I can switch to real channel numbers.,

**Comment by RogerSelwyn** [comment]
Yep it uses channel numbers. Not able to lookup based on favourites.

**Comment by B-Hartley** [comment]
no worries, love it. Thanks.

**Comment by B-Hartley** [comment]
Do you have a "tip jar" or a "buy me a coffee" or anything like that set up?    Would be happy to bung you a few quid to show my appreciation?

**Comment by RogerSelwyn** [comment]
No I don't, but thanks. Maybe when I stop having a proper job and need some pin money I'll set something up :-)

**Comment by B-Hartley** [comment]
no worries.

**Comment by RogerSelwyn** [comment]
I'll wait for a few days to see if we get any problems, and if anyone else who downloads complains... Then I'll release fully.
